### ISSUE TITLE
Implement Metrics

### ISSUE THREAD

**Issue by WGierke** [HPI-DeepLearning/SegMed#18]
BRATS uses 
- [x] Dice score
- [x] Hausdorff distance
- [x] Sensitivity
- [x] Specificity

- [x] Unify metric interface and output metric each training step and each sample step
- [ ] Output metric of all/x images in val every y epochs
We want to use that for a better evaluation of our model.

**Comment by WGierke** [comment]
To get started:
- [Dice Score](https://stackoverflow.com/questions/31273652/how-to-calculate-dice-coefficient-for-measuring-accuracy-of-image-segmentation-i)
- [Hausdorff Distance](https://github.com/sapphire008/Python/blob/master/generic/HausdorffDistance.py)
- [Sensitivity](https://radiopaedia.org/articles/sensitivity-and-specificity) = TP/(TP+FN) (measures the ovelap)
- [Specificity](https://radiopaedia.org/articles/sensitivity-and-specificity) = TN/(TN+FP) (counterpart of Sensitivity)

**Comment by ratnim** [comment]
-Wiki for [Dice Score](https://en.wikipedia.org/wiki/S%C3%B8rensen%E2%80%93Dice_coefficient)
